### ISSUE TITLE
Multiple video streams muxed in the same MKV file are not handled / selectable

### ISSUE THREAD

**Issue by Delgan** [jellyfin/jellyfin-web#1421]
**Describe the bug**
It's possible for a MKV file to contain several video tracks. However, the web client only shows one.

**To Reproduce**
1. Create a `.mkv` with multiple video tracks (eg. `ffmpeg -i input1.mp4 -i input2.mp4 -c copy -map 0 -map 1 output.mkv`)
2. Add the video file to the library
3. Go to the item details page on the web client
4. Notice that you can't select the video track to play

**Expected behavior**
All video tracks contained in the `.mkv` should be listed by the selector and usable by the player.

**System (please complete the following information):**
 - OS: Linux
 - Browser: Firefox
 - Jellyfin Version: 10.5.4

**Additional context**
Some remarks after discussion on Matrix:
* The selection [is purposefully disabled](https://github.com/jellyfin/jellyfin-web/blob/b57c6cacb805fc1bf10e34a4d5677c4f0c94295b/src/controllers/itemDetails.js#L195), it's been like this for years
* The selected video track is ignored in [the `play()` options](https://github.com/jellyfin/jellyfin-web/blob/b57c6cacb805fc1bf10e34a4d5677c4f0c94295b/src/controllers/itemDetails.js#L1952), thus enabling the `select` has no effect anyway
* It is therefore likely that it is also a problem related to the server not being able to handle multiple video streams

**Comment by dominikh** [comment]
Shut up, bot.

**Comment by itsTyrion** [comment]
anything? This would be useful for shows where creating a sub folder with multiple videos for every episode (I don't want/need 4k or so on mobile) is kinda chaotic/confusing(and VERY hit or miss)

**Comment by Thalhammer** [comment]
> This would be super useful for shows.

Not really. All audio/video streams use the same "namespace" if you can call it that way. Jellyfin would have no clue which audio stream is related to which video stream. However it would make pretranscoding content to lower resolutions/bitrates/nonhdr much more painless because you don't have manage multiple files, don't need to group the videos in jellyfin and you don't need to store audio streams multiple times for each video. It would be even cooler if jellyfin could autoselect the best matching video stream depending on device caps, but thats a different issue.

It might be stupid to preserve all audio streams in the pretranscoded copy but I like to aim for the best quality possible (which is why I am CPU transcoding beforehand in the first place) and transcoding audio on the fly is trivial even on the cpu. In some cases the audio streams end up taking a significant chunk of the files.
For example one movie, source 4K HEVC HDR (total 51GB, video 42GB, audio 8GB), the lowest bitrate 1080p copy I have is a 1080p H264 SDR version and now its total 11G, video 3G, audio 8G. Meaning on this single movie I'd save ~8G, which is a ~13% reduction. If I extrapolate that to my entire media collection we are talking TB's of saved space with 0 quality or performance reduction.

TLDR: Supporting this would be a game changer for me.

**Comment by Thalhammer** [comment]
> In which case you can still use separate files.

Oh yeah I misread your reply, I though you wanna cram all the episodes of a season into a single file.

> How many audio streams at what format and bitrate are you using that you have 8GB of audio in 1 movie??

Nothing particularly special:
- DTS-HD MA GER (3.6Mbit)
- DTS GER (1.5Mbit)
- DTS-HD MA JPN (3.6Mbit)
- DTS JPN (1.5Mbit)
But I do also have movies where I have 5-8 audio tracks in various languages. I tend to keep languages, even if I don't speak them because I might try to learn them at some point, in which case I'll happily watch them trying to understand as much as possible without looking at the subs.

**Comment by itsTyrion** [comment]
DTS-HD MA (MasterAudio) is bit-for-bit lossless AND it can have a sampling rate of 96kHz (for 7.1) and 24-bit - both of which you're likely not able to discern, especially the sampling rate > 48k For now, you could play with that maybe?

**Comment by Thalhammer** [comment]
> For now, you could play with that maybe?

I could, but I don't wanna waste high quality source material. I might when I run out of space at some point. It's not that I need the space, its just that not wasting space for something thats perfectly achievable with a (likely small) change in jellyfin sounds like a good idea. I might take a look at implementing it at some point, but I have not worked on jellyfins codebase before and it has also been a while since I last used c#, so no idea how far thats gonna go.

**Comment by wieluk** [comment]
Would be nice I have some presentation that i need to watch and they have 2 video streams one for the presenter and one for the presentation.

**Comment by Continous** [comment]
To piggyback on the point of storing multiple qualities in the same file, it would be fairly useful if you are less concerned about storage space, and more about unnecessarily transcoding things, so if you could store multiple codecs in the same file, and let Jellyfin choose whichever the client could directplay, that'd be really nice.

**Comment by dhouck** [comment]
> Would be nice I have some presentation videos that i need to watch and they have 2 video streams one for the presenter and one for the presentation.

Yeah, the few things I’ve found that do support multiple video tracks often don’t support this (you need to choose *one* track; you can’t see both at once).  VLC handles it in a particulasly weird way; it supports both but chooses whether it shows one track or many based on the file type (eg. an mkv you choose one track; an mp4 you get multiple at once).

It would be nice if Jellyfin could handle both uses cases, but even handling one or the other would be a good start.

**Comment by KirkMunro** [comment]
I just came across a good example where I'm trying to set this up as I digitize my personal movie/tv collection: Logan. My Logan blu-ray package comes with both Logan and Logan noir blu-ray discs. I'm using MKVToolNix to create a single mkv file with both video tracks, because it just makes sense. The chapters, audio, and subtitles are the same, because it's just the same movie, with an alternate video track (black and white vs color), so why hold on to all of that duplicated content and have multiple titles showing in the UX when it's just one movie. IMHO better to have a single mkv file with a choice of video tracks to allow the viewer to choose the one that they want.

**Comment by thornbill** [comment]
Closing in favor of this [feature request](https://features.jellyfin.org/posts/942/support-multi-track-video-quality)

**Comment by dhouck** [comment]
There are many things here *and even in that feature request* which are not equivalent to just choosing a specific quality video stream.  Some things here in particular are about playing multiple video tracks at once.
